apps/src/code-studio/makerlab/setupPage.js

```diff
@@ -0,0 +1,207 @@
+        return bc.ensureAppInstalled()
+            .then(() => this.succeed('app-installed'))
+            .catch((error) => this.fail('app-installed'));
+      }).then(() => {
+        this.spin('board-plug');
+        return BoardController.getDevicePort()
+            .then(() => this.succeed('board-plug'))
+            .catch((error) => this.fail('board-plug'));
+      }).then(() => {
+        this.spin('board-connect');
+        return bc.ensureBoardConnected()
+            .then(() => this.succeed('board-connect'))
+            .catch((error) => this.fail('board-connect'));
+      }).then(() => {
+        this.spin('board-components');
+        return bc.connectWithComponents()
+            .then(() => this.thumb('board-components'))
+            .then(() => {
+              bc.prewiredComponents.buzzer.play({
+                song: [
+                  ["G3", 100], ["C4", 100], ["E4", 100], ["G4", 50],
+                  [null, 150], ["E4", 75], ["G4", 400]
+                ], tempo: 45000
+              });
+              bc.prewiredComponents.colorLeds.forEach(l => l.color('green'));
+            })
+            .then(() => promiseWaitFor(1600))
+            .then(() => {
+              bc.prewiredComponents.colorLeds.forEach(l => l.off());
+              bc.prewiredComponents.buzzer.play({
+                song: [["C4", 100]],
+                tempo: 45000
+              });
+            })
+            .then(() => this.succeed('board-components'))
+            .catch((error) => this.fail('board-components'));
+      });
+    } else {
+      this.fail('is-chrome');
+    }
+
+    if (!isWindows()) {
+      this.hide('windows-drivers');
+    }
+  }
+});
+
+const SetupStep = React.createClass({
+  propTypes: {
+    stepName: React.PropTypes.string.isRequired,
+    stepId: React.PropTypes.string.isRequired,
+    stepStatus: React.PropTypes.oneOf(STEP_STATUSES).isRequired
+  },
+
+  iconFor(stepStatus) {
+    switch (stepStatus) {
+      case WAITING:
+        return "fa fa-fw fa-clock-o";
+      case ATTEMPTING:
+        return "fa fa-fw fa-spinner fa-spin";
+      case SUCCEEDED:
+        return "fa fa-fw fa-check-circle";
+      case CELEBRATING:
+        return "fa fa-fw fa-thumbs-o-up";
+      case FAILED:
+        return "fa fa-fw fa-times-circle";
+      default:
+        throw new Error('Unknown step status.');
+    }
+  },
+
+  styleFor(stepStatus) {
+    switch (stepStatus) {
+      case ATTEMPTING:
+      case WAITING:
+        return {color: '#949ca2'};
+      case SUCCEEDED:
+      case CELEBRATING:
+        return {color: 'green'};
+      case HIDDEN:
+        return {display: 'none'};
+      default:
+        return {color: 'red', fontWeight: 'bold'};
+    }
+  },
+
+  render() {
+    if (this.props.stepStatus === HIDDEN) {
+      return null;
+    }
+    return (
+        <div id={this.props.stepId}
+             style={Object.assign({margin: '15px 0'}, this.styleFor(this.props.stepStatus))}>
```

Comment:
We've been using [Radium](https://github.com/FormidableLabs/radium) for style compositions like this, which lets you pass an array of style objects as the style property.

```jsx
// Abbreviated
const SetupStep = Radium(React.createClass({
  render() {
    return (
      <div style={[{margin: '15px 0'}, this.styleFor(this.props.stepStatus)]}>
        {/* ... */}
      </div>
    );
  }
}));
```